packages/core/admin/ee/server/validation/review-workflows.js

```diff
@@ -15,6 +16,19 @@ const validateUpdateStageOnEntity = yup
   })
   .required();
 
+const validateContentTypes = yup.array().of(
+  yup.string().test({
+    name: 'content-type-exists',
+    message: (value) => `Content type ${value.originalValue} does not exist`,
+    test(uid) {
+      const model = strapi.getModel(uid);
+      if (!model) return false;
+      // It's not a valid  content type if it's not visible in the content manager
+      return getVisibleContentTypesUID({ [uid]: model }).includes(uid);
```

Comment:
we normally do this kind of validations in yup, also, wouldn't it be also using strapi global in the service? 
I find it handy to use yup, as it clearly marks which path is the one having issues.

Reply:
That's really handy indeed 🤔 
Isn't the service loaded like a factory? Shouldn't it be using the factory pattern? Did I miss something when doing the service? 🤔

Reply:
In this specific package we create it like:
```
const getService = (name, { strapi } = { strapi: global.strapi }) => {
  return strapi.service(`admin::${name}`);
};
```
The strapi `service` container does not create a factory of the service but returns previously initizliaed services that has stored in a hashmap.

```
import { Strapi } from '../../';

export type Service = {
  [key: string]: (...args: any) => any;
};

export type ServiceFactory = ({ strapi: Strapi }) => Service;
```

And our services itself are using the strapi global too 😓 . The idea is to import from singletons in the future rather than having a global .
We could revise this so we dont use the global though :) wdyt?

Reply:
oh, I see in review workflows we do not use the global. ignore my comment :D

Reply:
would you prefer not using the global but losing the yup validation error format?

Reply:
Lets keep it in the yup validation, maybe we can add a comment 
```suggestion
      // Warning; we use the strapi global - to avoid that, it would need to refactor how we generate validation function by using a factory with the strapi instance as parameter.
      const model = strapi.getModel(uid);
      if (!model) return false;
      // It's not a valid  content type if it's not visible in the content manager
      return getVisibleContentTypesUID({ [uid]: model }).includes(uid);
```